topics/go/concurrency/channels/example1/example1.go

```diff
@@ -147,8 +149,8 @@ func fanOutSem() {
 	emps := 20
 	ch := make(chan string, emps)
 
-	const cap = 5
-	sem := make(chan bool, cap)
+	g := runtime.NumCPU()
```

Comment:
I wouldn't name this variable `g` because it's not a count of goroutines. It's the size of the semaphore. So I guess it's a count of how many goroutines are doing the critical section at once but it seems likely to confuse people.

Reply:
We are specically setting the size of the semaphore to the number of threads we have. I think for pooling we want to teach to use this number as a default until you learn better. It is consistent with other use.

Reply:
Yeah, no I get that. I'm just saying `g` doesn't seem like the best name for the variable to me.

Reply:
Honestly, it is the only variable since you specifically want it to represent the number of goroutines you can run without the need for a context-switch. The idea here is to set the number based on the number of goroutines you want to use.

Reply:
Ok